client/src/main/java/org/opensearch/ml/client/MachineLearningNodeClient.java

```diff
@@ -182,6 +186,18 @@ public void registerModelGroup(
         client.execute(MLRegisterModelGroupAction.INSTANCE, mlRegisterModelGroupRequest, listener);
     }
 
+    /**
+     * Execute metrics correlation algorithm
+     * @param name Function name
+     * @param metricsCorrelationInput metrics correlation input
+     * @param listener a listener to be notified of the result
+     */
+    @Override
+    public void executeAlgorithm(FunctionName name, Input metricsCorrelationInput, ActionListener<MLExecuteTaskResponse> listener) {
```

Comment:
From here it seems like we are opening up `execute` api only for metrics correlation algorithm? Should we open up `execute api` in general (the way we open up train/predict api for all algorithms)? May be not tied to only metrics correlation algorithm?

@ylwu-amzn what do you think?

Reply:
There's no execute api in general. Already synced up with yaliang on this solution offline.

Reply:
I see, if that's the case can we add more comment and TODO in the code?

Reply:
@jackiehanyang I think we have some miscommunication here. We only have metrics correlation now, but you should not name the parameter name as `metricsCorrelationInput` here, that's too specific. Suggest change to a more general name like `input` 
```suggestion
    public void executeAlgorithm(FunctionName name, Input input, ActionListener<MLExecuteTaskResponse> listener) {
```

Reply:
updated the parameter name here - https://github.com/opensearch-project/ml-commons/pull/1540/files